File: fixtures/scale/typescript/large/src/registry/RegistryAffine.ts
import { CheckAffine } from '../checks/CheckAffine';
import { CheckDecimation } from '../checks/CheckDecimation';
import { CheckPromoter } from '../checks/CheckPromoter';
import { CheckModwheel } from '../checks/CheckModwheel';
import { CheckLinearmix } from '../checks/CheckLinearmix';
import { CheckSkewmap } from '../checks/CheckSkewmap';

export class RegistryAffine {
  static checks(): Array<[string, () => boolean]> {
    return [
      ['affine', CheckAffine.run],
      ['decimation', CheckDecimation.run],
      ['promoter', CheckPromoter.run],
      ['modwheel', CheckModwheel.run],
      ['linearmix', CheckLinearmix.run],
      ['skewmap', CheckSkewmap.run],
    ];
  }
}

File: fixtures/scale/lua/large/registry/registry_affine.lua
local check_affine = require("checks.check_affine")
local check_decimation = require("checks.check_decimation")
local check_promoter = require("checks.check_promoter")
local check_modwheel = require("checks.check_modwheel")
local check_linearmix = require("checks.check_linearmix")
local check_skewmap = require("checks.check_skewmap")

local function affine_checks()
  return {
    { "affine", check_affine.check_affine },
    { "decimation", check_decimation.check_decimation },
    { "promoter", check_promoter.check_promoter },
    { "modwheel", check_modwheel.check_modwheel },
    { "linearmix", check_linearmix.check_linearmix },
    { "skewmap", check_skewmap.check_skewmap },
  }
end

return { affine_checks = affine_checks }
